client-java/src/main/java/uk/gov/gchq/palisade/client/java/JavaClient.java

```diff
@@ -0,0 +1,233 @@
+/*
+ * Copyright 2020 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.gov.gchq.palisade.client.java;
+
+import io.micronaut.runtime.event.annotation.EventListener;
+import org.slf4j.*;
+
+import uk.gov.gchq.palisade.client.java.download.*;
+import uk.gov.gchq.palisade.client.java.job.*;
+import uk.gov.gchq.palisade.client.java.request.*;
+import uk.gov.gchq.palisade.client.java.resource.*;
+import uk.gov.gchq.palisade.client.java.util.Bus;
+
+import javax.inject.Singleton;
+import javax.websocket.ContainerProvider;
+
+import java.net.URI;
+import java.util.*;
+import java.util.function.UnaryOperator;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import static uk.gov.gchq.palisade.client.java.job.IJobContext.createJobContext;
+
+/**
+ * <p>
+ * The main client implementation.
+ * </p>
+ * <p>
+ * Note that this class should only be created via the static {@code create}
+ * methods on the {@link Client} interface. For tests it can be injected into a
+ * test class annotated with {@code @MicronautTest} as it is configured for
+ * dependency injection. This makes testing much easier.
+ *
+ * @author dbell
+ * @since 0.5.0
+ */
+@Singleton
+public class JavaClient implements Client {
+
+    private static final Logger log = LoggerFactory.getLogger(JavaClient.class);
```

Comment:
```suggestion
    private static final Logger LOGGER = LoggerFactory.getLogger(JavaClient.class);
```
and other appropriate changes

Reply:
Done as part of the code smells.